oslabs-beta/DockerLocal
# Login button fix

## Description:
#### Problems:
-  Login button required user to click button multiple times to allow user to enter. 
- User was displayed as the user's username instead of their real Github Username.
- Toggle button is still being used to simulate login feature
- tslint errors/console logs throughout codebase

#### Solution:
- Login issue was due to incorrect parsing of access token. Cookie sometimes comes in with username rather than token first. Conditional statements were added to verify and correctly parse the token.
- Username was passed as state to correct component to display correct username
- Toggle button was removed
- Console logs were removed and some code modified to satisfy tslint

## Changes I Made:
- `index.ts`: add a void to createWindow function 
- `app.tsx`: remove running app console log
- `cookieClientHelper.ts`: Fix cookie parsing in helper function
- `routes.tsx`: adds a checkifLoggedIn function that will verify accessToken and allow for login conditional rendering to properly function. Also remove dummy dev login button.
- `home.tsx`: disable es-lint no unsresolved
- `fetchRepohelper.tsx`: make comments ts-lint compliant and change lets to consts
- `authController.tsx`: make comments ts-lint compliant and add void to function
- `dockerController.ts`: make comments ts-lint compliant


## How to Test:
- npm install
- npm run start
- Click` Sign in with Github` button ONCE
- See your username displayed on the top left of the projects page.
<img width="780" alt="Screen Shot 2020-07-11 at 10 48 48 AM" src="https://user-images.githubusercontent.com/32973305/87230316-1bdf8200-c364-11ea-8890-8ae43bde4836.png">
